pkg/api/controller.go

```diff
@@ -2936,9 +2936,9 @@ func (c *Controller) handleAPIErrorCallback(ctx context.Context, w http.Response
 
 	// Handle Hook Errors
 	var hookAbortErr *graveler.HookAbortError
-	if errors.As(err, &hookAbortErr) {
-		log.WithField("run_id", hookAbortErr.RunID).Warn("aborted by hooks")
-		cb(w, r, http.StatusPreconditionFailed, hookAbortErr.Unwrap())
+	if errors.As(err, &hookAbortErr) && errors.Is(hookAbortErr, actions.ErrActionFailed) {
```

Comment:
note that we have a specific error for action failed, we can add a case in the switch to return predicate failed for this error and remove this code.